tests/integration/test_automation/testParser.py

```diff
@@ -0,0 +1,507 @@
+        log_file.write("WARNING !! TB ignores addressses in files not specified in Yaml via start/end indexso if Yaml does not include all addresses, it will still flag it as pass !!! \n")
+
+def are_there_any_extra_messages():
+
+  
+    is_there_extra_messages = False;
+    log_file.write("Checking for extra messages \n")
+    for theName, theObject in list_checker_dict.items():
+        log_file.write(theName + "\n")
+        if theObject.isListEmpty() != True:
+            log_file.write(theName + " has extra messages \n" )
+            is_there_extra_messages = True
+
+
+    return is_there_extra_messages        
+
+    
+
+if __name__ == "__main__":
+   buffer_access_dict = {}
+   filenames_dict = {}
+   #arsnoop dictionary defs from from IHI0022H_c_amba_axi_protocol_spec.pdf table D3.7
+   arsnoop_dict = {'ReadNoSnoop':0,'ReadOnce':0,'ReadShared':1,'ReadClean':1,'ReadNotSharedDirty':3,'ReadUnique':7,'CleanUnique':11,'CleanShared':8,'CleanInvalid':9,'MakeInvalid':13}
+   #awsnoop dictionary defs from from IHI0022H_c_amba_axi_protocol_spec.pdf table D3.8
+   awsnoop_dict = {'WriteNoSnoop':0,'WriteUnique':0,'WriteLineUnique':1,'WriteClean':2,'WriteBack':3,'Evict':4,'WriteEvict':5}
+   #acsnoop dictionary defs from from IHI0022H_c_amba_axi_protocol_spec.pdf table D3.19
+   acsnoop_dict = {'ReadOnce':0,'ReadShared':1,'ReadClean':2,'ReadNotSharedDirty':3,'ReadUnique':7,'CleanShared':8,'CleanInvalid':9}   
+   ace_master_0_read = []   
+   ace_master_1_read = []
+   ace_master_2_read = []
+   ace_master_3_read = []
+   ace_master_0_write  = []
+   ace_master_1_write  = []
+   ace_master_2_write  = []
+   ace_master_3_write  = []
+   ace_master_0_snoop = []
+   ace_master_1_snoop  = []
+   ace_master_2_snoop  = []
+   ace_master_3_snoop = []
+   axi_slave_read  = []
+   axi_slave_write  = []
+   buffer_address_list = []
+  
+
+   print(Style.RESET_ALL)
+ 
+   with open('test.yaml', 'r') as file:
+       test_yaml = yaml.safe_load(file)
+       print("Test: " + test_yaml['test']['test_name'])
+
+   with open('common.yaml', 'r') as file2:
+       test_common_yaml = yaml.safe_load(file2)
+
+       
+   file1 = open('main.map', 'r')
+   log_file = open(test_yaml['test']['test_name'] + "_test_result.log", 'w')
+   Lines = file1.readlines()       
+
+   #skip analysis of test if test sequence marks it so
+   if "test_action" in test_yaml['test']:
+       if test_yaml['test']['test_action'] == "skip":
+          print(Fore.YELLOW + "Test Skipped")
+          if "test_notes" in test_yaml['test']:
+              print(Fore.YELLOW + test_yaml['test']['test_notes'])
+              log_file.write(test_yaml['test']['test_notes'])
+              log_file.close()
+              file1.close()
+          print(Style.RESET_ALL)
+          exit(0)
+       
+
+   #create a dictionary of checkers so that the right one can be acquried by key from the test sequence.
+   list_checker_dict = {}
+   RACE = ACE_LIST_CHECKER(ace_master_0_read,  arsnoop_dict, 'ARSNOOP')
```

Comment:
could be rolled in a for loop

Reply:
yeah I do something similar in the get_and_filter_data. So that would mean there are three loops.

Reply:
There is also an error here where for axi is axi_master_.., instead of axi_slave_. These keys are used in the test sequences. I should fix this

Reply:
I started to do this but then found that the code to put it into a loop was a bit cumbersome and outweighed its' benefit. This is partly because there are still globals scattered around like the lists ace_master_0_read, which some functions make use of it. I would say that we get it in now , make use of it and then open up another ticket to remove all globals.

Reply:
OK